core/src/test/java/org/apache/iceberg/TestDeleteFiles.java

```diff
@@ -268,6 +268,40 @@ public void testCannotDeleteFileWhereNotAllRowsMatchPartitionFilter() {
             .commit());
   }
 
+  @Test
+  public void testDeleteCaseSensitivity() {
+    table.newFastAppend()
+        .appendFile(DATA_FILE_BUCKET_0_IDS_0_2)
+        .commit();
+
+    Expression rowFilter = Expressions.lessThan("iD", 5);
+
+    AssertHelpers.assertThrows("Should use case sensitive binding by default",
```

Comment:
Just curious, does it make sense to make non case sensitive the default?

Reply:
Case sensitive is more strict, so I think that makes sense as the default.

Reply:
Iceberg historically uses case sensitivity resolution by default in all other places.